### PR TITLE
Update kessel criterion for email integration

### PR DESCRIPTION
**Pull request by g-duval**
Name and structure updates from ADR-53 reviews

### CODE REVIEW

File: engine/src/main/java/com/redhat/cloud/notifications/processors/ExternalAuthorizationCriterion.java [hunk 1]
```diff
@@ -3,34 +3,35 @@
...
 import java.util.Objects;
 
 @JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
 @JsonIgnoreProperties(ignoreUnknown = true)
-public class ExternalAuthorizationCriteria {
+public class ExternalAuthorizationCriterion {
 
     @NotNull
-    private String assetType;
+    private JsonObject type;
```
Comment (MikelAlejoBR) [inline engine/src/main/java/com/redhat/cloud/notifications/processors/ExternalAuthorizationCriterion.java:15]: Why not have a dedicated object for the type? I think we will have to modify the `getLookupSubjectsRequest` to adjust the `.setName()` and the `.setNamespace()` calls.

**Comment by MikelAlejoBR** [comment]
Also, I think we need to change the subject type to `rbac/principal` too: https://github.com/RedHatInsights/notifications-backend/blob/ae2f54ca6df5de2a79377c0e1ffa52065d9e126b/recipients-resolver/src/main/java/com/redhat/cloud/notifications/recipients/resolver/kessel/KesselService.java#L60

File: engine/src/main/java/com/redhat/cloud/notifications/processors/ExternalAuthorizationCriterion.java (hunk 1 above)
Comment (g-duval) [reply engine/src/main/java/com/redhat/cloud/notifications/processors/ExternalAuthorizationCriterion.java]: I was still dreaming to be agnostic of Kessel structure, but you're right, we have to be realistic :)

File: recipients-resolver/src/main/java/com/redhat/cloud/notifications/recipients/resolver/kessel/KesselService.java [hunk 2]
```diff
@@ -21,6 +21,8 @@ public class KesselService {
 
     static final String SUBJECT_TYPE_USER = "user";
```
Comment (MikelAlejoBR) [inline recipients-resolver/src/main/java/com/redhat/cloud/notifications/recipients/resolver/kessel/KesselService.java:22]: This one needs to be `principal`: https://github.com/RedHatInsights/rbac-config/blob/97d7501a55cff790dc44162a6ede8412be2528a8/configs/prod/schemas/schema.zed#L13

**Review by MikelAlejoBR — APPROVED** [review]
LGTM!

**Pull request merged**